californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -268,7 +275,12 @@ public void onReadyToSend() {
 					// original request so that at the end of the
 					// blockwise transfer the Matcher can correctly
 					// close the overall exchange
-					request.setToken(block.getToken());
+					if (request.getToken() == null) {
```

Comment:
Which case do we try to  support here ?

Reply:
The case, where the token is supplied by the application. I don't think, that overwriting makes sense, if the token is already set.

Reply:
OK